src/compat/eme/set_media_keys.ts

```diff
@@ -0,0 +1,51 @@
+import log from "../../log";
+import sleep from "../../utils/sleep";
+import shouldAwaitSetMediaKeys from "../should_await_set_media_keys";
+import { ICustomMediaKeys } from "./custom_media_keys";
+import { IEmeApiImplementation } from "./eme-api-implementation";
+
+/**
+ * @param {Object} emeImplementation
+ * @param {Object} mediaElement
+ * @param {Object|null} mediaKeys
+ * @returns {Promise}
+ */
+export function setMediaKeys(
+  emeImplementation : IEmeApiImplementation,
+  mediaElement : HTMLMediaElement,
+  mediaKeys : MediaKeys | ICustomMediaKeys | null
+): Promise<unknown> {
+  if (shouldAwaitSetMediaKeys()) {
+    return emeImplementation.setMediaKeys(mediaElement, mediaKeys)
+      .then(() => {
+        log.info("Compat: MediaKeys updated with success");
+      });
+  }
+
+  return Promise.race(
+    [
+      emeImplementation.setMediaKeys(mediaElement, mediaKeys)
+        .then(() => {
+          log.info("Compat: MediaKeys updated with success");
+        })
+        .catch(err => {
+          if (mediaKeys === null) {
+            log.error(
+              "Compat: Could not reset MediaKeys",
+              err instanceof Error ? err : "Unknown Error"
+            );
+            return;
```

Comment:
Could you provide some clarification on why a return statement is used on this case, while an error is thrown for the case "Could not update" ?

Reply:
Failing to ""disable"" MediaKeys (setting the HTMLMediaElement `mediaKeys` property to `null` is much less problematic than failing to set one, because it just mean that you may be able to decrypt content you didn't intend to.

As EME API are very often badly implemented, we let this one pass.